lib/string_reverse.rb

```diff
@@ -1,6 +1,46 @@
+require "pry"
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of the string passed into the string_reverse method.
+# Space complexity: Space complexity is O(1) because the amount of space occupied by the characters in the string plus each of the variables does not change.
+
 def string_reverse(my_string)
-  raise NotImplementedError
+  i_start = 0
+  i_end = -1
+  a = ""
+  z = ""
+  t = ""
+  loop_times = my_string.length

```

Comment:
You probably want to use a number less than `my_string.length`. Try printing out `my_string` and `loop_times` in each iteration of the loop below?